src/LowerTriangularMatrices/lower_triangular_matrix.jl

```diff
@@ -1,117 +1,184 @@
+Base.one(L::LowerTriangularArray{T,N,AT}) where {T,N,AT} = zeros(LowerTriangularArray{T,N,AT}, size(L)...)
 
-function LowerTriangularMatrix{T}(::UndefInitializer, m::Integer, n::Integer) where T
-    return LowerTriangularMatrix(Vector{T}(undef, nonzeros(m, n)), m, n)
+function LowerTriangularArray{T,N,ArrayType}(::UndefInitializer, I::Vararg{Integer,N}) where {T,N,ArrayType<:AbstractArray{T}}
+    return LowerTriangularArray(ArrayType(undef, nonzeros(I[1], I[2]), I[3:end]...), I[1], I[2])
 end
 
-Base.randn(::Type{LowerTriangularMatrix{T}}, m::Integer, n::Integer) where T = LowerTriangularMatrix(randn(T, nonzeros(m, n)), m, n)
-Base.rand(::Type{LowerTriangularMatrix{T}}, m::Integer, n::Integer) where T = LowerTriangularMatrix(rand(T, nonzeros(m, n)), m, n)
+function LowerTriangularMatrix{T,ArrayType}(::UndefInitializer, m::Integer, n::Integer) where {T,ArrayType<:AbstractArray{T}}
+    return LowerTriangularMatrix(ArrayType(undef, nonzeros(m, n)), m, n)
+end
+
+Base.randn(::Type{LowerTriangularArray{T}}, m::Integer, n::Integer, I::Vararg{Integer,N}) where {T,N} = LowerTriangularArray(randn(T, nonzeros(m, n), I...), m, n)
+Base.rand(::Type{LowerTriangularArray{T}}, m::Integer, n::Integer, I::Vararg{Integer,N}) where {T,N} = LowerTriangularArray(randn(T, nonzeros(m, n), I...), m, n)
 
 Base.randn(::Type{LowerTriangularMatrix}, m::Integer, n::Integer) = LowerTriangularMatrix(randn(nonzeros(m, n)), m, n)
 Base.rand(::Type{LowerTriangularMatrix}, m::Integer, n::Integer) = LowerTriangularMatrix(rand(nonzeros(m, n)), m, n)
 
+Base.randn(::Type{LowerTriangularMatrix{T}}, m::Integer, n::Integer) where T = LowerTriangularMatrix(randn(T, nonzeros(m, n)), m, n)
+Base.rand(::Type{LowerTriangularMatrix{T}}, m::Integer, n::Integer) where T = LowerTriangularMatrix(rand(T, nonzeros(m, n)), m, n)
+
 # INDEXING
 """
-    k = ij2k(   i::Integer,     # row index of matrix
-                j::Integer,     # column index of matrix
-                m::Integer)     # number of rows in matrix
-
+$(TYPEDSIGNATURES)
 Converts the index pair `i, j` of an `m`x`n` LowerTriangularMatrix `L` to a single
 index `k` that indexes the same element in the corresponding vector that stores
 only the lower triangle (the non-zero entries) of `L`."""
 @inline ij2k(i::Integer, j::Integer, m::Integer) = i+(j-1)*m-j*(j-1)÷2
 triangle_number(n::Integer) = n*(n+1)÷2
 nonzeros(m::Integer, n::Integer) = m*n-triangle_number(n-1)
 
-@inline function Base.getindex(L::LowerTriangularMatrix, k::Integer)
-    @boundscheck 0 < k <= length(L.data) || throw(BoundsError(L, k))
-    @inbounds r = L.data[k]
+"""
+$(TYPEDSIGNATURES)
+Converts the linear index `k` in the lower triangle into a pair `(i, j)` of indices 
+of the matrix in column-major form. (Formula taken from 
+Angeletti et al, 2019, https://hal.science/hal-02047514/document)
+"""
+@inline function k2ij(k::Integer, m::Integer) 
+    kp = triangle_number(m) - k 
+    p = Int(floor((sqrt(1 + 8*kp) - 1)/2))
```

Comment:
I can't find it's use?

Reply:
I added it for the broadcast and then didn't use it so far, but yes it would need to be precomputed for any actual test, and yes we could use it in some tests as well.